### ISSUE TITLE
The linter and formatter results are inconsistent

### ISSUE THREAD

**Issue by m15o** [pinterest/ktlint#854]
## Expected Behavior

Trying to format the following code. Expected no errors.
```kotlin
package test

class PropertyA
class PropertyB
class PropertyC

fun someFunction(a: PropertyA, b: PropertyB, c: PropertyC): String = TODO()

fun main(args: Array<String>) {
    if (someFunction(
        PropertyA(),
        PropertyB(),
        PropertyC()
    ) ==
        "ABC"
    ) {
        TODO()
    }
}
```

## Observed Behavior

ktlintFormat task fails with the following message
```
> Task :ktlintMainSourceSetFormat FAILED
/Users/me/IdeaProjects/test/src/main/kotlin/main.kt:11:13: Unexpected indentation (expected 8, actual 12)
/Users/me/IdeaProjects/test/src/main/kotlin/main.kt:12:17: Unexpected indentation (expected 8, actual 12)
/Users/me/IdeaProjects/test/src/main/kotlin/main.kt:14:1: Unexpected indentation (expected 8, actual 12)
/Users/me/IdeaProjects/test/src/main/kotlin/main.kt:15:12: Unexpected indentation (expected 4, actual 8)
```

ktlintCheck task produces following errors:
```
> Task :ktlintMainSourceSetCheck FAILED
/Users/me/IdeaProjects/test/src/main/kotlin/main.kt:11:1: Unexpected indentation (8) (should be 12)
/Users/me/IdeaProjects/test/src/main/kotlin/main.kt:12:1: Unexpected indentation (8) (should be 12)
/Users/me/IdeaProjects/test/src/main/kotlin/main.kt:13:1: Unexpected indentation (8) (should be 12)
/Users/me/IdeaProjects/test/src/main/kotlin/main.kt:14:1: Unexpected indentation (4) (should be 8)
```

It seem to be formatted in continuation inden 8:
```kotlin
package test

class PropertyA
class PropertyB
class PropertyC

fun someFunction(a: PropertyA, b: PropertyB, c: PropertyC): String = TODO()

fun main(args: Array<String>) {
    if (someFunction(
            PropertyA(),
            PropertyB(),
            PropertyC()
        ) ==
        "ABC"
    ) {
        TODO()
    }
}
```

## Your Environment
* Version of ktlint used: 0.38.1
* Used ktlint gradleplugin (org.jlleitschuh.gradle.ktlint:9.3.0
* Version of Gradle used (if applicable): 6.3
* Operating System and version: macOS 10.15.6
